core/src/apps/management/apply_settings.py

```diff
@@ -11,6 +11,16 @@
     from trezor.messages.ApplySettings import ApplySettings
 
 
+def humanize_time(milliseconds: int) -> str:
+    units = (("hour", 60 * 60 * 1000), ("minute", 60 * 1000), ("second", 1000))
+    for (u, divisor) in units:
+        if milliseconds >= divisor:
+            num = milliseconds // divisor
+            return "{} {}{}".format(num, u, "s" if num > 1 else "")
```

Comment:
Neat! I've rewrote the function slightly and moved it to `trezor/strings.py`, let me know if it's the wrong thing to do since it's not used from anywhere else.

Reply:
yeah, that's a good idea